File: Java/src/com/leetcode/Medium/MinimumNumberOfStepsToMakeTwoStringsAnagram.java
package com.leetcode.Medium;

// 5333. Minimum Number of Steps to Make Two Strings Anagram
public class MinimumNumberOfStepsToMakeTwoStringsAnagram {
    public int minSteps(String s, String t) {
        int count = 0;
        int[] charCount = new int[26];

        for (int i = 0; i < s.length(); i++)
            charCount[s.charAt(i) - 'a']++;

        for (int i = 0; i < t.length(); i++) {
            if (charCount[t.charAt(i) - 'a']-- <= 0)
                count++;
        }

        return count;
    }
}

File: JavaScript/Medium/MinimumNumberOfStepsToMakeTwoStringsAnagram.js
/**
 * 1347. Minimum Number of Steps to Make Two Strings Anagram
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
const minSteps = function (s, t) {
  let count = 0;
  let charCount = new Array(26).fill(0);

  for (let i = 0; i < s.length; i++)
    charCount[s.charCodeAt(i) - 'a'.charCodeAt(0)]++;

  for (let i = 0; i < t.length; i++) {
    if (charCount[t.charCodeAt(i) - 'a'.charCodeAt(0)]-- <= 0)
      count++;
  }

  return count;
};